common/darwin/Classes/FlutterRTCDesktopCapturer.m

```diff
@@ -0,0 +1,295 @@
+BOOL _captureWindow = NO;
+BOOL _captureScreen = NO;
+NSArray<RTCDesktopSource *>* _captureSources;
+#endif
+
+@implementation FlutterWebRTCPlugin (DesktopCapturer)
+
+-(void)getDisplayMedia:(NSDictionary *)constraints
+                result:(FlutterResult)result {
+    NSString *mediaStreamId = [[NSUUID UUID] UUIDString];
+    RTCMediaStream *mediaStream = [self.peerConnectionFactory mediaStreamWithStreamId:mediaStreamId];
+    RTCVideoSource *videoSource = [self.peerConnectionFactory videoSource];
+
+#if TARGET_OS_IPHONE
+ BOOL useBroadcastExtension = false;
+    id videoConstraints = constraints[@"video"];
+    if ([videoConstraints isKindOfClass:[NSDictionary class]]) {
+       // constraints.video.deviceId
+        useBroadcastExtension = [((NSDictionary *)videoConstraints)[@"deviceId"] isEqualToString:@"broadcast"];
+    }
+    
+    id screenCapturer;
+    
+    if(useBroadcastExtension){
+        screenCapturer = [[FlutterBroadcastScreenCapturer alloc] initWithDelegate:videoSource];
+    } else {
+        screenCapturer = [[FlutterRPScreenRecorder alloc] initWithDelegate:videoSource];
+    }
+    
+    [screenCapturer startCapture];
+    NSLog(@"start %@ capture", useBroadcastExtension ? @"broadcast" : @"replykit");
+        
+    self.videoCapturerStopHandlers[mediaStreamId] = ^(CompletionHandler handler) {
+        NSLog(@"stop %@ capture", useBroadcastExtension ? @"broadcast" : @"replykit");
+        [screenCapturer stopCaptureWithCompletionHandler:handler];
+    };
+
+    if(useBroadcastExtension) {
+        NSString *extension = [[[NSBundle mainBundle] infoDictionary] valueForKey: kRTCScreenSharingExtension];
+        if(extension) {
+            RPSystemBroadcastPickerView *picker = [[RPSystemBroadcastPickerView alloc] init];
+            picker.preferredExtension = extension;
+            picker.showsMicrophoneButton = false;
+            
+            SEL selector = NSSelectorFromString(@"buttonPressed:");
+            if([picker respondsToSelector:selector]) {
+                [picker performSelector:selector withObject:nil];
+            }
+        }
+    }
+#endif
+    
+#if TARGET_OS_OSX
+/* example for constraints:
+    {
+        'audio': false,
+        'video": {
+            'deviceId':  {'exact': sourceId},
+            'mandatory': {
+                'frameRate': 30.0
+            },
+        }
+    }
+*/
+    NSString *sourceId = nil;
+    NSInteger fps = 30;
+    id videoConstraints = constraints[@"video"];
+    if ([videoConstraints isKindOfClass:[NSDictionary class]]) {
+         NSDictionary *deviceId = videoConstraints[@"deviceId"];
+        if (deviceId != nil && [deviceId isKindOfClass:[NSDictionary class]]) {
+            if(deviceId[@"exact"] != nil) {
+                sourceId = deviceId[@"exact"];
+                if(sourceId == nil) {
+                    result(@{@"error": @"No sourceId found"});
+                    return;
+                }
+            }
+        }
```

Comment:
Should we fall back to default screen if no deviceId is provided ?
I'm not sure of the specs.